### PR TITLE
Fix HERON python path usage

### LINKED ISSUE #181: [TASK] HERON should not need to add src to python path
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
It would be useful (and needed if we create a PIP package) if HERON could use just plugins as the path.

**Describe the solution you'd like**
This requires changing all imports to add HERON.src

**Describe alternatives you've considered**
Adding this hack to RAVEN:
```patch
From 7a79a7c40575a3413522ab20763d4a2e075a5e90 Mon Sep 17 00:00:00 2001
From: "Joshua J. Cogliati" <joshua.cogliati@inl.gov>
Date: Tue, 3 May 2022 16:39:42 -0600
Subject: [PATCH] Hard coding paths for HERON.

---
 ravenframework/CustomDrivers/DriverUtils.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ravenframework/CustomDrivers/DriverUtils.py b/ravenframework/CustomDrivers/DriverUtils.py
index 9787b9ba4e..16ff404853 100644
--- a/ravenframework/CustomDrivers/DriverUtils.py
+++ b/ravenframework/CustomDrivers/DriverUtils.py
@@ -108,6 +108,12 @@ def setupCpp():
   else:
     ##TODO REMOVE PP3 WHEN RAY IS AVAILABLE FOR WINDOWS
     utils.add_path_recursively(os.path.join(frameworkDir,'contrib','pp'))
+  #XXX Needed for HERON on cluster
+  ravenDir = os.path.dirname(frameworkDir)
+  heronSrc = os.path.join(ravenDir, "plugins", "HERON", "src")
+  if os.path.exists(heronSrc) and heronSrc not in sys.path:
+    sys.path.append(heronSrc)
+    sys.path.append(os.path.join(ravenDir, "scripts"))
 
 def checkVersions():
   """
```

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #181 

##### What are the significant changes in functionality due to this change request?
This changes HERON to using import HERON.src.whatever from import whatever
This also makes DispatchManager a ExternalModel plugin.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: src/DispatchManager.py [hunk 1]
```diff
@@ -15,12 +15,13 @@
 from typing_extensions import final
 
 # set up path
-sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
-import _utils as hutils
-import SerializationManager
+#sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
```
Comment (dylanjm) [inline src/DispatchManager.py]: Should we just go ahead and remove this comment?

File: templates/template_driver.py [hunk 2]
```diff
@@ -700,7 +700,10 @@ def _modify_inner(self, template, case, components, sources):
     input_filepath = os.path.abspath((os.path.dirname(__file__)))
     input_filepath = input_filepath+'/../src/DispatchManager'
     ext_model = template.find('Models').find('ExternalModel')
-    ext_model.set('ModuleToLoad', input_filepath)
+    #ext_model.set('ModuleToLoad', input_filepath)
```
Comment (dylanjm) [inline templates/template_driver.py]: Remove commented code.

**Review by dylanjm — CHANGES_REQUESTED** [review]
Just minor change requests. Other than that it should be ready to merge.

File: src/DispatchManager.py (hunk 1 above)
Comment (joshua-cogliati-inl) [reply src/DispatchManager.py]: Fixed.

File: templates/template_driver.py (hunk 2 above)
Comment (joshua-cogliati-inl) [reply templates/template_driver.py]: Fixed.

**Review by dylanjm — APPROVED** [review]
All changes look good. Will merge once all tests pass.

**Pull request merged**